app/lib/api_integration/settings_result.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+class ApiIntegration::SettingsResult
+  extend ActiveModel::Naming
+  extend ActiveModel::Translation
+  class ServiceMismatchError < ActiveRecord::ActiveRecordError; end
+
+  def initialize(service:, proxy:)
+    raise ServiceMismatchError unless proxy.service == service
+    @service = service
+    @proxy = proxy
+  end
+
+  attr_reader :service, :proxy
+
+  def valid?
+    # It's done this way to get all the errors even if it is already known that is invalid
+    [service.valid?, proxy.valid?].all?
+  end
+
+  def update!(service_attributes: {}, proxy_attributes: {})
+    ActiveRecord::Base.transaction do
+      service.update!(service_attributes)
+      proxy.update!(proxy_attributes) unless service.deployment_option == 'hosted'
+    end
+    true
+  end
+
+  def update(service_attributes: {}, proxy_attributes: {})
+    update!(service_attributes: service_attributes, proxy_attributes: proxy_attributes)
+    # Done this way so it does the rollback
+  rescue ActiveRecord::RecordInvalid
+    false
+  end
+
+  def errors
+    errors = ActiveModel::Errors.new(self)
+    proxy.errors.full_messages.each    { |proxy_error|   errors.add(:proxy, proxy_error)     }
+    service.errors.full_messages.each  { |service_error| errors.add(:service, service_error) }
+    errors
+  end
+
+  private
+
+  # ActiveModel::Errors needs this method to read the errors correctly
+  def read_attribute_for_validation(attr)
+    public_send(attr)
+  end
+end
```

Comment:
why to put this in `lib` ? 🤔 

I think it is better to put it in `app/models` or `app/services`.. what do you think?